### PR TITLE
(doc): git-based bundles with custom directory specified 

### PR DESCRIPTION
**Pull request by exdx**
This PR adds additional documentation for bundles with custom bundle directory roots.

### CODE REVIEW

File: docs/git-bundles.md [hunk 1]
```diff
@@ -8,8 +8,9 @@ make it available within the cluster.
 
 The git repository backing the bundle, or "bundle git repo", must have a certain structure in order to produce a valid
 plain Bundle. It should have a directory where the Kubernetes manifests are stored -- this directory is used when
-copying manifests onto the cluster. By default, this directory is assumed to be `/manifests` at the top level. For an
-example of a bundle git repo that conforms to the specifications, see
+copying manifests onto the cluster. By default, this directory is assumed to be `/manifests` at the root level, but can
+be specified via `spec.source.git.directory`. There must be a `manifests` directory in the provided directory in order to
+have a valid bundle git repo. For an example of a bundle git repo that conforms to the specifications, see
 the [combo repository](https://github.com/operator-framework/combo/).
```
Comment (timflannagan) [inline docs/git-bundles.md:14]: How do we feel about making this a note to make it more explicit:

---

The git repository backing the bundle, or "bundle git repo", must have a certain structure in order to produce a valid
plain Bundle. It should have a directory where the Kubernetes manifests are stored -- this directory is used when
copying manifests onto the cluster. By default, this directory is assumed to be `/manifests` at the root level, but can
be specified via `spec.source.git.directory`.

**Note**: There must be a `manifests` directory in the provided directory in order to
have a valid bundle git repo.

For an example of a bundle git repo that conforms to the specifications, see
the [combo repository](https://github.com/operator-framework/combo/).

**Comment by timflannagan** [comment]
Should we separate out the doc and e2e changes in their own separate commits?

File: docs/git-bundles.md (hunk 1 above)
Comment (exdx) [reply docs/git-bundles.md:14]: That sounds good

**Comment by exdx** [comment]
> Should we separate out the doc and e2e changes in their own separate commits?

We could yes, especially since the e2e is not passing currently (can't figure out why). This PR can just be for the doc then.

**Review by timflannagan — APPROVED** [review]

**Comment by joelanford** [comment]
@exdx @timflannagan 

> The git repository backing the bundle, or "bundle git repo", must have a certain structure in order to produce a valid plain Bundle. It should have a directory where the Kubernetes manifests are stored -- this directory is used when copying manifests onto the cluster.

Should the "Git-based bundles" doc talk about plain bundles other than potentially using them as an example? It seems like we're conflating how a bundle is sourced (`bundle.spec.source`) and how a bundle is parsed/interpreted (`bundle.spec.provisionerClassName`)

> By default, this directory is assumed to be /manifests at the root level, but can be specified via spec.source.git.directory.

This ^ text isn't quite aligned to the implementation, still. I think this should be:

> By default, this directory is assumed to be git repository root, but can be specified via `spec.source.git.directory`. 
> 
> Take plain+v0 bundles for example. In plain+v0 bundles, we expect to find `./manifests` in the bundle root directory. When sourcing the plain+v0 bundle from a git repository, the bundle root is defined by `spec.source.git.directory` (which defaults to `<gitRoot>` when unset). Therefore, when `spec.source.git.directory` is unset, we expect to find `<gitRoot>/manifests`. When `spec.source.git.directory` is set to `path/to/bundle`, we expect to find `<gitRoot>/path/to/bundle/manifests`.

**Comment by exdx** [comment]
> @exdx @timflannagan
> 
> > The git repository backing the bundle, or "bundle git repo", must have a certain structure in order to produce a valid plain Bundle. It should have a directory where the Kubernetes manifests are stored -- this directory is used when copying manifests onto the cluster.
> 
> Should the "Git-based bundles" doc talk about plain bundles other than potentially using them as an example? It seems like we're conflating how a bundle is sourced (`bundle.spec.source`) and how a bundle is parsed/interpreted (`bundle.spec.provisionerClassName`)
> 
> > By default, this directory is assumed to be /manifests at the root level, but can be specified via spec.source.git.directory.
> 
> This ^ text isn't quite aligned to the implementation, still. I think this should be:
> 
> > By default, this directory is assumed to be git repository root, but can be specified via `spec.source.git.directory`.
> > Take plain+v0 bundles for example. In plain+v0 bundles, we expect to find `./manifests` in the bundle root directory. When sourcing the plain+v0 bundle from a git repository, the bundle root is defined by `spec.source.git.directory` (which defaults to `<gitRoot>` when unset). Therefore, when `spec.source.git.directory` is unset, we expect to find `<gitRoot>/manifests`. When `spec.source.git.directory` is set to `path/to/bundle`, we expect to find `<gitRoot>/path/to/bundle/manifests`.

Based on the implementation we have currently, the `spec.source.git.directory` has to point to the manifests directly to have the unpack operation succeed. So if the manifests are in `./dev/deploy/manifests` then the directory variable has to point to there (see #238 for an example). It's almost as `spec.source.git.directory` should be renamed to `spec.source.git.manifestsDir` based on what our code is doing currently. I'm going to follow-up with some additional documentation improvements to make this more clear, now that the custom directory test is passing.

**Comment by timflannagan** [comment]
@exdx @joelanford Should we open a ticket for making the documentation agnostic to an individual provisioner?

**Comment by exdx** [comment]
We could -- I'm addressing this in a follow-up for this doc now

**Comment by joelanford** [comment]
I _think_ #240  PR resolves the issue with how `spec.source.git.directory` is behaving.

**Pull request merged**